Correct month last day when changing date panel dates

SergioCrisostomo opened the pull request:
Correct date navigation when starting from the last day of a month that has more days than the next chosen month.

Fixes #3773

SergioCrisostomo committed:
Correct month last day

icarusion commented:
When select date, the interval between the left and right panels has become 2 months,
the current month is 2018.5(left), 2018.6(right), now, the right is 2018.7

![image](https://user-images.githubusercontent.com/5370542/44375036-2c926f00-a524-11e8-921c-f594f0e4f738.png)

SergioCrisostomo committed:
Use first of month and not other panels date

since in cases where prev month date > next month date it will jump one more month ahead

SergioCrisostomo commented:
@icarusion thanks, updated
